### PR TITLE
Print warning if pack excludes a file because of default exclude wildcards

### PR DESCRIPTION
**Pull request by rohit21agrawal**
Fixes:  https://github.com/NuGet/Home/issues/3308

### CODE REVIEW

File: src/NuGet.Core/NuGet.Commands/Strings.resx [hunk 1]
```diff
@@ -691,4 +691,7 @@ For more information, visit http://docs.nuget.org/docs/reference/command-line-re
   <data name="Error_InvalidDependencyVersionConstraints" xml:space="preserve">
     <value>Package version constraints for '{0}' return a version range that is empty.</value>
   </data>
+  <data name="Warning_FileExcludedByDefault" xml:space="preserve">
+    <value>File '{0}' was not added as NuGet package files and files and folders starting with a dot are excluded by default. To prevent this, use -NoDefaultExcludes from the command line.</value>
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Commands/Strings.resx]: How about a more terse:

`File '{0}' was not added to the nupkg. Files and folders starting with '.' are excluded by default. To include this file use -NoDefaultExcludes`

File: src/NuGet.Core/NuGet.Commands/Strings.resx [hunk 2]
```diff
@@ -694,4 +694,7 @@ For more information, visit http://docs.nuget.org/docs/reference/command-line-re
   <data name="Error_UnableToSignPackage" xml:space="preserve">
     <value>Unable to sign package.</value>
   </data>
+  <data name="Warning_FileExcludedByDefault" xml:space="preserve">
+    <value>File '{0}' was not added to the package. Files and folders starting with '.' or ending with '.nupkg' are excluded by default. To include this file, use -NoDefaultExcludes from the commandline</value>
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Commands/Strings.resx:698]: This will only hit for nuspec pack right?
Like the -NoDefaultExcludes makes sense in nuspec pack?

**Review by nkolev92 — APPROVED** [review]
LGTM

File: src/NuGet.Core/NuGet.Commands/Strings.resx (hunk 2 above)
Comment (rohit21agrawal) [reply src/NuGet.Core/NuGet.Commands/Strings.resx:698]: No this applies to all pack scenarios

Comment (pakrym) [reply src/NuGet.Core/NuGet.Commands/Strings.resx:698]: What about MSBuild task? How do you disable NoDefaultExcludes in CSPROJ?

Comment (rohit21agrawal) [reply src/NuGet.Core/NuGet.Commands/Strings.resx:698]: @pakrym you are right. that doesn't exist right now, and I just realized you filed a bug for it. I'll see if I can get this enabled for 15.7 

https://github.com/NuGet/Home/issues/6450

**Pull request merged**